library/core/src/panic/location.rs

```diff
@@ -126,7 +141,17 @@ impl<'a> Location<'a> {
     #[rustc_const_unstable(feature = "const_location_fields", issue = "102911")]
     #[inline]
     pub const fn file(&self) -> &str {
-        self.file
+        #[cfg(bootstrap)]
+        {
+            self.file
+        }
+        #[cfg(not(bootstrap))]
+        // SAFETY: Codegen ensures that there is a null byte.
+        unsafe {
+            let cstr = CStr::from_ptr(self.file as _);
+            let len = cstr.count_bytes();
+            crate::str::from_utf8_unchecked(crate::slice::from_raw_parts(self.file as _, len))
```

Comment:
Another option would be doing a varint length prefix instead of a nul terminator. For short paths it would only be 1 byte so the same size as a nul terminator.

Reply:
Are these *ever* on the hot path?

Reply:
Logging, maybe?